tool/generate_code.sh

```diff
@@ -3,15 +3,17 @@
 # Copyright 2022 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
-
+TOOL_DIR=`dirname "${BASH_SOURCE[0]}"`
```

Comment:
What is BASH_SOURCE[0]? Where it is documented?

Can you add explanation in a comment?

Reply:
BASH_SOURCE[0] essentially gives a path the the script, relative to where it was called from.

i've added some comments that hopefully help explain this :)

Reply:
https://stackoverflow.com/a/35006505/464770
Is where I learned about it :)